File: Racing/v1/Racing.java
import java.awt.*;
import java.awt.event.*;
import java.util.Random;
import javax.swing.*;

public class Racing extends JFrame {
    public Racing() {
        add(new RacingPanel());
        setTitle("Need for Speed Style Java Racing");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setResizable(false);
        pack();
        setLocationRelativeTo(null);
        setVisible(true);
    }

    public static void main(String[] args) {
        new Racing();
    }
}

class RacingPanel extends JPanel implements ActionListener, KeyListener {
    private final int WIDTH = 500;
    private final int HEIGHT = 700;
    private Timer timer = new Timer(16, this);

    private int carX = 225;
    private int carY = 560;
    private int carW = 50;
    private int carH = 90;

    private int speed = 6;
    private int roadOffset = 0;
    private int score = 0;

    private boolean left, right, up, down;
    private boolean gameOver = false;

    private Random rand = new Random();
    private Rectangle[] traffic = {
            new Rectangle(120, -100, 50, 90),
            new Rectangle(330, -350, 50, 90),
            new Rectangle(220, -600, 50, 90)
    };

    public RacingPanel() {
        setPreferredSize(new Dimension(WIDTH, HEIGHT));
        setBackground(Color.BLACK);
        addKeyListener(this);
        setFocusable(true);
        timer.start();
    }

    protected void paintComponent(Graphics g) {
        super.paintComponent(g);

        drawRoad(g);
        drawTraffic(g);
        drawPlayer(g);
        drawHUD(g);

        if (gameOver) {
            g.setColor(Color.WHITE);
            g.setFont(new Font("Arial", Font.BOLD, 36));
            g.drawString("CRASH!", 185, 330);
            g.setFont(new Font("Arial", Font.BOLD, 20));
            g.drawString("Press SPACE to restart", 145, 365);
        }
    }

    private void drawRoad(Graphics g) {
        g.setColor(Color.GRAY);
        g.fillRect(70, 0, 360, HEIGHT);

        g.setColor(Color.WHITE);
        g.fillRect(70, 0, 8, HEIGHT);
        g.fillRect(422, 0, 8, HEIGHT);

        for (int y = -80 + roadOffset; y < HEIGHT; y += 120) {
            g.fillRect(245, y, 10, 70);
        }
    }

    private void drawPlayer(Graphics g) {
        g.setColor(Color.BLUE);
        g.fillRect(carX, carY, carW, carH);
        g.setColor(Color.CYAN);
        g.fillRect(carX + 8, carY + 10, carW - 16, 18);
    }

    private void drawTraffic(Graphics g) {
        g.setColor(Color.RED);
        for (Rectangle r : traffic) {
            g.fillRect(r.x, r.y, r.width, r.height);
        }
    }

    private void drawHUD(Graphics g) {
        g.setColor(Color.WHITE);
        g.setFont(new Font("Arial", Font.BOLD, 20));
        g.drawString("Score: " + score, 20, 30);
        g.drawString("Speed: " + speed, 370, 30);
    }

    public void actionPerformed(ActionEvent e) {
        if (!gameOver) {
            updateCar();
            updateRoad();
            updateTraffic();
            checkCollisions();
            score++;
        }

        repaint();
    }

    private void updateCar() {
        if (left) carX -= 6;
        if (right) carX += 6;
        if (up && speed < 14) speed++;
        if (down && speed > 4) speed--;

        if (carX < 80) carX = 80;
        if (carX > 370) carX = 370;
    }

    private void updateRoad() {
        roadOffset += speed;
        if (roadOffset >= 120) roadOffset = 0;
    }

    private void updateTraffic() {
        for (Rectangle r : traffic) {
            r.y += speed;

            if (r.y > HEIGHT) {
                r.y = -rand.nextInt(500) - 100;
                int[] lanes = {120, 225, 330};
                r.x = lanes[rand.nextInt(lanes.length)];
            }
        }
    }

    private void checkCollisions() {
        Rectangle player = new Rectangle(carX, carY, carW, carH);

        for (Rectangle r : traffic) {
            if (player.intersects(r)) {
                gameOver = true;
            }
        }
    }

    private void restart() {
        carX = 225;
        speed = 6;
        score = 0;
        gameOver = false;

        traffic[0].setLocation(120, -100);
        traffic[1].setLocation(330, -350);
        traffic[2].setLocation(220, -600);
    }

    public void keyPressed(KeyEvent e) {
        if (e.getKeyCode() == KeyEvent.VK_LEFT) left = true;
        if (e.getKeyCode() == KeyEvent.VK_RIGHT) right = true;
        if (e.getKeyCode() == KeyEvent.VK_UP) up = true;
        if (e.getKeyCode() == KeyEvent.VK_DOWN) down = true;

        if (e.getKeyCode() == KeyEvent.VK_SPACE && gameOver) {
            restart();
        }
    }

    public void keyReleased(KeyEvent e) {
        if (e.getKeyCode() == KeyEvent.VK_LEFT) left = false;
        if (e.getKeyCode() == KeyEvent.VK_RIGHT) right = false;
        if (e.getKeyCode() == KeyEvent.VK_UP) up = false;
        if (e.getKeyCode() == KeyEvent.VK_DOWN) down = false;
    }

    public void keyTyped(KeyEvent e) {}
}

File: Racing/v1/Racing.py
import tkinter as tk
import random

WIDTH, HEIGHT = 500, 700

class Racing:
    def __init__(self):
        self.root = tk.Tk()
        self.root.title("Need for Speed Style Python Racing")
        self.canvas = tk.Canvas(self.root, width=WIDTH, height=HEIGHT, bg="black")
        self.canvas.pack()

        self.car_x = 225
        self.car_y = 560
        self.car_w = 50
        self.car_h = 90

        self.speed = 6
        self.road_offset = 0
        self.score = 0
        self.game_over = False

        self.keys = set()
        self.traffic = [
            [120, -100, 50, 90],
            [330, -350, 50, 90],
            [225, -600, 50, 90]
        ]

        self.root.bind("<KeyPress>", self.key_press)
        self.root.bind("<KeyRelease>", self.key_release)

        self.loop()
        self.root.mainloop()

    def key_press(self, e):
        self.keys.add(e.keysym.lower())

        if e.keysym == "space" and self.game_over:
            self.restart()

    def key_release(self, e):
        self.keys.discard(e.keysym.lower())

    def update(self):
        if self.game_over:
            return

        if "left" in self.keys:
            self.car_x -= 6
        if "right" in self.keys:
            self.car_x += 6
        if "up" in self.keys and self.speed < 14:
            self.speed += 1
        if "down" in self.keys and self.speed > 4:
            self.speed -= 1

        self.car_x = max(80, min(370, self.car_x))

        self.road_offset += self.speed
        if self.road_offset >= 120:
            self.road_offset = 0

        for car in self.traffic:
            car[1] += self.speed
            if car[1] > HEIGHT:
                car[1] = -random.randint(100, 600)
                car[0] = random.choice([120, 225, 330])

        self.check_collisions()
        self.score += 1

    def intersects(self, a, b):
        ax, ay, aw, ah = a
        bx, by, bw, bh = b
        return ax < bx + bw and ax + aw > bx and ay < by + bh and ay + ah > by

    def check_collisions(self):
        player = [self.car_x, self.car_y, self.car_w, self.car_h]

        for car in self.traffic:
            if self.intersects(player, car):
                self.game_over = True

    def restart(self):
        self.car_x = 225
        self.speed = 6
        self.score = 0
        self.game_over = False
        self.traffic = [
            [120, -100, 50, 90],
            [330, -350, 50, 90],
            [225, -600, 50, 90]
        ]

    def draw(self):
        self.canvas.delete("all")

        self.canvas.create_rectangle(70, 0, 430, HEIGHT, fill="gray")
        self.canvas.create_rectangle(70, 0, 78, HEIGHT, fill="white")
        self.canvas.create_rectangle(422, 0, 430, HEIGHT, fill="white")

        for y in range(-80 + self.road_offset, HEIGHT, 120):
            self.canvas.create_rectangle(245, y, 255, y + 70, fill="white")

        for car in self.traffic:
            x, y, w, h = car
            self.canvas.create_rectangle(x, y, x + w, y + h, fill="red")

        self.canvas.create_rectangle(self.car_x, self.car_y, self.car_x + self.car_w, self.car_y + self.car_h, fill="blue")
        self.canvas.create_rectangle(self.car_x + 8, self.car_y + 10, self.car_x + self.car_w - 8, self.car_y + 28, fill="cyan")

        self.canvas.create_text(70, 30, text=f"Score: {self.score}", fill="white", font=("Arial", 18, "bold"))
        self.canvas.create_text(420, 30, text=f"Speed: {self.speed}", fill="white", font=("Arial", 18, "bold"))

        if self.game_over:
            self.canvas.create_text(WIDTH // 2, HEIGHT // 2, text="CRASH!", fill="white", font=("Arial", 40, "bold"))
            self.canvas.create_text(WIDTH // 2, HEIGHT // 2 + 40, text="Press SPACE to restart", fill="white", font=("Arial", 18, "bold"))

    def loop(self):
        self.update()
        self.draw()
        self.root.after(16, self.loop)

Racing()
